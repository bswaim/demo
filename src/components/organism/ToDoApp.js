import React, {useEffect, useState} from 'react';
import AddListItem from "../atom/AddListItem";
import {
    addNewItemToList,
    moveCurrentState,
    removeCheckedItems, updateLatestChangeToDoHistory
} from "../../utils/utils";
import CheckListSection from "../molecule/CheckListSection";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {
    faCheckDouble,
    faListCheck,
    faRotateLeft,
    faRotateRight,
    faTrash,
    faTrashAlt
} from "@fortawesome/free-solid-svg-icons";
import {find} from "lodash";

function ToDoApp() {
    const [listHistoryState, setListHistoryState] = useState([{isCurrent: true, list: [], isInEditMode: false}]);
    const currentListView = find([...listHistoryState], x => x.isCurrent)?.list || [];

    const undoEnabledLogic = !([...listHistoryState][0]?.isCurrent); // if first item in list is not current view
    const redoEnabledLogic = !([...listHistoryState][listHistoryState.length -1]?.isCurrent); // if last item in list's isCurrent prop is NOT true


    useEffect(() => {
        //TEST
        console.log('listHistoryState changed!');
        console.log('listHistoryState');
        console.log(listHistoryState);
        // console.log('currentListView');
        // console.log(currentListView);

    }, [listHistoryState]);

    const deleteAllItems = () => {
        const updateCurrentState = updateLatestChangeToDoHistory([], listHistoryState)
        setListHistoryState([...updateCurrentState])
    }
    const deleteAllCompletedItems = () => {
        const updatedArray = removeCheckedItems([...currentListView]);
        const updateCurrentState = updateLatestChangeToDoHistory(updatedArray, listHistoryState)
        setListHistoryState([...updateCurrentState])
    }
    const undoListStateChange = () => {
        const updateCurrentState = moveCurrentState(listHistoryState);
        setListHistoryState([...updateCurrentState]);
    }
    const redoListStateChange = () => {
        // setUndoRedoInProgress(true);

        const updateCurrentState = moveCurrentState(listHistoryState, false);
        setListHistoryState([...updateCurrentState]);
    }

    return (
        <div id='todo-id' className='todo-container bg-primary-bg max-w-2xl rounded-2xl border-2 border-solid border-black p-4'>
            <h1 className='text-center text-xl border-b-2 border-b-primary border-solid pb-2 pl-4 flex justify-between'>
                <div>To-Do App</div>
                {/* DELETE ALL ITEMS BUTTON */}
                <button onClick={() => deleteAllItems()}>
                    <FontAwesomeIcon icon={faTrash} className='text-delete-color'/>
                    <FontAwesomeIcon icon={faListCheck} className='text-delete-color ml-1'/>
                </button>
            </h1>

            {/* TO DO ITEMS START */}
            <CheckListSection
                listHistoryState={listHistoryState}
                setListHistoryState={(e) => setListHistoryState([...e])} // need to update list
            />
            <div className='text-left'>
                <AddListItem onClicked={() => {
                    const updatedArray = addNewItemToList(currentListView);
                    const updatedListHistory = updateLatestChangeToDoHistory(updatedArray, listHistoryState, true);
                    setListHistoryState([...updatedListHistory]);
                }}/>
            </div>

            {/* SEPARATOR */}
            <div className='w-full h-2 border-b-2 border-b-primary border-dashed mb-4'/>

            {/* COMPLETED SECTION START */}
            <h2 className='text-left ml-6 flex justify-between'>
                COMPLETED
                <button onClick={() => deleteAllCompletedItems()}>
                    <FontAwesomeIcon icon={faTrashAlt} className='text-delete-color'/>
                    <FontAwesomeIcon icon={faCheckDouble} className='text-delete-color ml-1'/>
                </button>
            </h2>
            <CheckListSection
                groupedByUnChecked
                listHistoryState={listHistoryState}
                setListHistoryState={(e) => setListHistoryState([...e])}
            />

            {/* UNDO BUTTON */}
            <div className='flex justify-between px-4 mt-8 text-2xl bg-white rounded-b-lg'>
                <button
                    id='undo'
                    className={undoEnabledLogic ? 'cursor-pointer' : 'cursor-default'}
                    onClick={() => undoListStateChange()}
                    disabled={!(undoEnabledLogic)}
                >
                    <FontAwesomeIcon icon={faRotateLeft} className={undoEnabledLogic ? 'text-secondary-text' : 'text-gray-500'}/>
                </button>

                {/* REDO BUTTON */}
                <button
                    id='redo'
                    className='cursor-pointer'
                    onClick={() => redoListStateChange()}
                    disabled={!(redoEnabledLogic)}
                >
                    <FontAwesomeIcon icon={faRotateRight} className={redoEnabledLogic ? 'text-secondary-text' : 'text-gray-500'}/>
                </button>
            </div>
        </div>
    );
}
export default ToDoApp;