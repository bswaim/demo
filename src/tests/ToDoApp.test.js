import {render, screen} from "@testing-library/react";
import ToDoApp from "../components/organism/ToDoApp";

describe('TO-DO APP', () => {
    test('default render components', () => {
        const component = render(<ToDoApp />);
        const container = screen.getByTestId(/todo-id/i);
        const header = screen.getByText(/to-do app/i);
        const deleteAllIcon = screen.getByTestId(/delete-all-icon/i);
        const deleteAllCheckedIcon = screen.getByTestId(/delete-all-checked-icon/i);
        const addNewItemIcon = screen.getByTestId(/add-new-icon/i);
        const addNewItemText = screen.getByTestId(/add-new-text/i);
        const completedText = screen.getByText(/Completed/i);
        const undoIcon = screen.getByTestId(/undo-icon/i);
        const redoIcon = screen.getByTestId(/redo-icon/i);

        expect(container).toBeInTheDocument();
        expect(header).toBeInTheDocument();
        expect(deleteAllIcon).toBeInTheDocument();
        expect(deleteAllCheckedIcon).toBeInTheDocument();
        expect(addNewItemIcon).toBeInTheDocument();
        expect(addNewItemText).toBeInTheDocument();
        expect(completedText).toBeInTheDocument();
        expect(undoIcon).toBeInTheDocument();
        expect(redoIcon).toBeInTheDocument();
    });
});