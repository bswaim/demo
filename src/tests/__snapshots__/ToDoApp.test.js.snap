// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TO-DO APP default render components 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todo-container bg-primary-bg max-w-2xl rounded-2xl border-2 border-solid border-black p-4"
        data-testid="todo-id"
        id="todo-id"
      >
        <h1
          class="text-center text-xl border-b-2 border-b-primary border-solid pb-2 pl-4 flex justify-between"
        >
          <div>
            To-Do App
          </div>
          <button
            data-testid="delete-all-icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-trash text-delete-color"
              data-icon="trash"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-list-check text-delete-color ml-1"
              data-icon="list-check"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M152.1 38.2c9.9 8.9 10.7 24 1.8 33.9l-72 80c-4.4 4.9-10.6 7.8-17.2 7.9s-12.9-2.4-17.6-7L7 113C-2.3 103.6-2.3 88.4 7 79s24.6-9.4 33.9 0l22.1 22.1 55.1-61.2c8.9-9.9 24-10.7 33.9-1.8zm0 160c9.9 8.9 10.7 24 1.8 33.9l-72 80c-4.4 4.9-10.6 7.8-17.2 7.9s-12.9-2.4-17.6-7L7 273c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l22.1 22.1 55.1-61.2c8.9-9.9 24-10.7 33.9-1.8zM224 96c0-17.7 14.3-32 32-32H480c17.7 0 32 14.3 32 32s-14.3 32-32 32H256c-17.7 0-32-14.3-32-32zm0 160c0-17.7 14.3-32 32-32H480c17.7 0 32 14.3 32 32s-14.3 32-32 32H256c-17.7 0-32-14.3-32-32zM160 416c0-17.7 14.3-32 32-32H480c17.7 0 32 14.3 32 32s-14.3 32-32 32H192c-17.7 0-32-14.3-32-32zM48 368a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                fill="currentColor"
              />
            </svg>
          </button>
        </h1>
        <ul />
        <div
          class="text-left"
        >
          <div
            class="flex"
          >
            <button
              class="text-2xl ml-4 m-2 pt-.5 pb-1 px-2 rounded font-bold"
              data-testid="add-new-icon"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-plus "
                data-icon="circle-plus"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM232 344V280H168c-13.3 0-24-10.7-24-24s10.7-24 24-24h64V168c0-13.3 10.7-24 24-24s24 10.7 24 24v64h64c13.3 0 24 10.7 24 24s-10.7 24-24 24H280v64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <h2
              class="mt-3 italic cursor-pointer"
              data-testid="add-new-text"
            >
              Add new item
            </h2>
          </div>
        </div>
        <div
          class="w-full h-2 border-b-2 border-b-primary border-dashed mb-4"
        />
        <h2
          class="text-left ml-6 flex justify-between"
        >
          COMPLETED
          <button
            data-testid="delete-all-checked-icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-trash-can text-delete-color"
              data-icon="trash-can"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-check-double text-delete-color ml-1"
              data-icon="check-double"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 178.7l-57.4-57.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l80 80c12.5 12.5 32.8 12.5 45.3 0l160-160zm96 128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 402.7 54.6 297.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l256-256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </h2>
        <ul />
        <div
          class="flex justify-between px-4 mt-8 text-2xl bg-white rounded-b-lg"
        >
          <button
            class="cursor-default"
            data-testid="undo-icon"
            disabled=""
            id="undo"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-rotate-left text-gray-500"
              data-icon="rotate-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48.5 224H40c-13.3 0-24-10.7-24-24V72c0-9.7 5.8-18.5 14.8-22.2s19.3-1.7 26.2 5.2L98.6 96.6c87.6-86.5 228.7-86.2 315.8 1c87.5 87.5 87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3c-62.2-62.2-162.7-62.5-225.3-1L185 183c6.9 6.9 8.9 17.2 5.2 26.2s-12.5 14.8-22.2 14.8H48.5z"
                fill="currentColor"
              />
            </svg>
          </button>
          <button
            class="cursor-default"
            data-testid="redo-icon"
            disabled=""
            id="redo"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-rotate-right text-gray-500"
              data-icon="rotate-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M463.5 224H472c13.3 0 24-10.7 24-24V72c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2L413.4 96.6c-87.6-86.5-228.7-86.2-315.8 1c-87.5 87.5-87.5 229.3 0 316.8s229.3 87.5 316.8 0c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0c-62.5 62.5-163.8 62.5-226.3 0s-62.5-163.8 0-226.3c62.2-62.2 162.7-62.5 225.3-1L327 183c-6.9 6.9-8.9 17.2-5.2 26.2s12.5 14.8 22.2 14.8H463.5z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todo-container bg-primary-bg max-w-2xl rounded-2xl border-2 border-solid border-black p-4"
      data-testid="todo-id"
      id="todo-id"
    >
      <h1
        class="text-center text-xl border-b-2 border-b-primary border-solid pb-2 pl-4 flex justify-between"
      >
        <div>
          To-Do App
        </div>
        <button
          data-testid="delete-all-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash text-delete-color"
            data-icon="trash"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M135.2 17.7L128 32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H320l-7.2-14.3C307.4 6.8 296.3 0 284.2 0H163.8c-12.1 0-23.2 6.8-28.6 17.7zM416 128H32L53.2 467c1.6 25.3 22.6 45 47.9 45H346.9c25.3 0 46.3-19.7 47.9-45L416 128z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-list-check text-delete-color ml-1"
            data-icon="list-check"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M152.1 38.2c9.9 8.9 10.7 24 1.8 33.9l-72 80c-4.4 4.9-10.6 7.8-17.2 7.9s-12.9-2.4-17.6-7L7 113C-2.3 103.6-2.3 88.4 7 79s24.6-9.4 33.9 0l22.1 22.1 55.1-61.2c8.9-9.9 24-10.7 33.9-1.8zm0 160c9.9 8.9 10.7 24 1.8 33.9l-72 80c-4.4 4.9-10.6 7.8-17.2 7.9s-12.9-2.4-17.6-7L7 273c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l22.1 22.1 55.1-61.2c8.9-9.9 24-10.7 33.9-1.8zM224 96c0-17.7 14.3-32 32-32H480c17.7 0 32 14.3 32 32s-14.3 32-32 32H256c-17.7 0-32-14.3-32-32zm0 160c0-17.7 14.3-32 32-32H480c17.7 0 32 14.3 32 32s-14.3 32-32 32H256c-17.7 0-32-14.3-32-32zM160 416c0-17.7 14.3-32 32-32H480c17.7 0 32 14.3 32 32s-14.3 32-32 32H192c-17.7 0-32-14.3-32-32zM48 368a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h1>
      <ul />
      <div
        class="text-left"
      >
        <div
          class="flex"
        >
          <button
            class="text-2xl ml-4 m-2 pt-.5 pb-1 px-2 rounded font-bold"
            data-testid="add-new-icon"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-plus "
              data-icon="circle-plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM232 344V280H168c-13.3 0-24-10.7-24-24s10.7-24 24-24h64V168c0-13.3 10.7-24 24-24s24 10.7 24 24v64h64c13.3 0 24 10.7 24 24s-10.7 24-24 24H280v64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
                fill="currentColor"
              />
            </svg>
          </button>
          <h2
            class="mt-3 italic cursor-pointer"
            data-testid="add-new-text"
          >
            Add new item
          </h2>
        </div>
      </div>
      <div
        class="w-full h-2 border-b-2 border-b-primary border-dashed mb-4"
      />
      <h2
        class="text-left ml-6 flex justify-between"
      >
        COMPLETED
        <button
          data-testid="delete-all-checked-icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash-can text-delete-color"
            data-icon="trash-can"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check-double text-delete-color ml-1"
            data-icon="check-double"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M342.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 178.7l-57.4-57.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l80 80c12.5 12.5 32.8 12.5 45.3 0l160-160zm96 128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 402.7 54.6 297.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l256-256z"
              fill="currentColor"
            />
          </svg>
        </button>
      </h2>
      <ul />
      <div
        class="flex justify-between px-4 mt-8 text-2xl bg-white rounded-b-lg"
      >
        <button
          class="cursor-default"
          data-testid="undo-icon"
          disabled=""
          id="undo"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-rotate-left text-gray-500"
            data-icon="rotate-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M48.5 224H40c-13.3 0-24-10.7-24-24V72c0-9.7 5.8-18.5 14.8-22.2s19.3-1.7 26.2 5.2L98.6 96.6c87.6-86.5 228.7-86.2 315.8 1c87.5 87.5 87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3c-62.2-62.2-162.7-62.5-225.3-1L185 183c6.9 6.9 8.9 17.2 5.2 26.2s-12.5 14.8-22.2 14.8H48.5z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="cursor-default"
          data-testid="redo-icon"
          disabled=""
          id="redo"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-rotate-right text-gray-500"
            data-icon="rotate-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M463.5 224H472c13.3 0 24-10.7 24-24V72c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2L413.4 96.6c-87.6-86.5-228.7-86.2-315.8 1c-87.5 87.5-87.5 229.3 0 316.8s229.3 87.5 316.8 0c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0c-62.5 62.5-163.8 62.5-226.3 0s-62.5-163.8 0-226.3c62.2-62.2 162.7-62.5 225.3-1L327 183c-6.9 6.9-8.9 17.2-5.2 26.2s12.5 14.8 22.2 14.8H463.5z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
